lms/static/js/learner_dashboard/models/program_subscription_model.js

```diff
@@ -0,0 +1,81 @@
+import Backbone from 'backbone';
+import moment from 'moment';
+
+import DateUtils from 'edx-ui-toolkit/js/utils/date-utils';
+
+/**
+ * Model for Program Subscription Data.
+ */
+class ProgramSubscriptionModel extends Backbone.Model {
+    constructor([data = {}], urls, userPreferences, ...args) {
+        const subscriptionState = data.subscription_state?.toLowerCase();
+        const subscriptionPrice =
+            data.currency === 'USD'
+                ? `$${data.price}`
+                : `${data.price} ${data.currency}`;
+        const subscriptionUrl =
+            subscriptionState === 'active'
+                ? urls.manage_subscription_url
+                : urls.buy_subscription_url;
+
+        const hasActiveTrial =
+            subscriptionState === 'active' && data.trial_end
+                ? moment(
+                    DateUtils.localizeTime(
+                        DateUtils.stringToMoment(data.trial_end),
+                        'UTC'
+                    )
+                ).isAfter(moment.utc())
+                : false;
+
+        const [nextPaymentDate] = ProgramSubscriptionModel.formatDate(
+            data.next_payment_date,
```

Comment:
I believe the same applies to `next_payment_date` except that it's not something the `/stripe-subscribe` endpoint sets

Reply:
Not an issue since this date is only needed when user has subscribed at which point it should have been set in the backend, before that its fine if its undefined/null as it won't be used anywhere